tsdb/head.go

```diff
@@ -805,6 +805,26 @@ func (h *Head) Init(minValidTime int64) error {
 	if err != nil && err != record.ErrNotFound {
 		return errors.Wrap(err, "find last checkpoint")
 	}
+
+	// Find the last segment.
+	// The purpose of this scope is to avoid wal.Segments err override wal.LastCheckpoint err.
+	var endAt int
+	{
+		var err error
+		_, endAt, err = wal.Segments(h.wal.Dir())
+		if err != nil {
+			return errors.Wrap(err, "finding WAL segments")
+		}
+	}
+
+	// Not likely endAt will smaller then startFrom,
+	// as the head cut a new wal segment file before creating a new checkpoint.
+	if endAt < startFrom {
+		endAt = startFrom
+	}
+
+	h.startWALReplayStatus(startFrom, endAt)
```

Comment:
I am not sure where is it actually considering the checkpoint. From what I understand, the status starts from `startFrom`, and even after checkpoint replay it is still `startFrom`. Only thing that changes is the startup screen comes up early?

Reply:
> From what I understand, the status starts from `startFrom`, and even after checkpoint replay it is still `startFrom`
What i understand is the `startFrom` will be `startFrom++` after the checkpoint replay. 
That's right At the front end, 

> the startup screen comes up early?